source/mainnet/smart-contracts/tutorials/wCCD/wCCD-interacting.rst

```diff
@@ -0,0 +1,778 @@
+
+or
+
+.. code-block:: console
+
+    $./concordium-client contract invoke IMPLEMENTATION --entrypoint operatorOf --parameter-json operatorOf.json --grpc-port 10001
+
+.. image:: ./images/wCCD_tutorial_7.png
+    :width: 100 %
+
+The ``tokenMetadata`` function
+==============================
+
+.. dropdown:: Input parameters for the ``tokenMetadata`` function (click here)
+
+    Create a ``tokenMetadata.json`` file and insert the following JSON array.
+
+    .. code-block:: json
+
+        [""]
+
+    .. note::
+
+        This empty string is required because of the CIS-2 token standard.
+        The `tokenId` of the wCCD token is the smallest unit possible (an empty string).
+
+You are ready now to invoke the ``tokenMetadata`` function with one of the following commands.
+
+.. code-block:: console
+
+    $./concordium-client contract invoke PROXY --entrypoint tokenMetadata --schema tokenMetadata_fallback_schema.bin --parameter-json tokenMetadata.json --grpc-port 10001
+
+or
+
+.. code-block:: console
+
+    $./concordium-client contract invoke IMPLEMENTATION --entrypoint tokenMetadata --parameter-json tokenMetadata.json --grpc-port 10001
+
+.. image:: ./images/wCCD_tutorial_8.png
+    :width: 100 %
+
+The ``supports`` function
+=========================
+
+.. dropdown:: Input parameters for the ``supports`` function (click here)
+
+    Create a ``supports.json`` file and insert the following example JSON array. It will query if
+    the two token standards (``CIS-0`` and ``CIS-2``) are supported by the wCCD token.
+
+    .. code-block:: json
+
+        ["CIS-0","CIS-2"]
+
+    .. note::
+
+        You can find more information about the `CIS-0 standard <https://proposals.concordium.software/CIS/cis-0.html>`_
+        and the `CIS-2 standard <https://proposals.concordium.software/CIS/cis-2.html>`_.
+
+You are ready now to invoke the ``supports`` function with one of the following commands.
+
+.. code-block:: console
+
+    $./concordium-client contract invoke PROXY --entrypoint supports --schema supports_fallback_schema.bin --parameter-json supports.json --grpc-port 10001
+
+or
+
+.. code-block:: console
+
+    $./concordium-client contract invoke IMPLEMENTATION --entrypoint supports --parameter-json supports.json --grpc-port 10001
+
+The below screenshot shows the response of querying if the wCCD
+token contract supports the following standards
+``["12345","CIS-0","CIS-2"]``. Its response is that it does
+not support the standard ``12345`` but it supports the ``CIS-0`` and the ``CIS-1`` standards.
+
+.. image:: ./images/wCCD_tutorial_9.png
+    :width: 100 %
+
+State-mutative functions
+------------------------
+
+The protocol has four state-mutative functions (``wrap``, ``unwrap``,
+``transfer``, and ``updateOperator``) that you can invoke on the ``proxy`` contract.
+These invokes will be passed through the fallback function on the ``proxy`` to the ``implementation`` contract.
```

Comment:
The `fallback function` has not been explained at this point.

Perhaps we could do something like this to de-emphasize the need to understand fallbacks, but still allow the user to find the info.

```suggestion
These invokes will be passed through the ``proxy`` to the ``implementation`` contract (via the `fallback entrypoint <...>`_).
```

Reply:
These invokes will be passed through the ``proxy`` to the ``implementation`` contract (via the `fallback entrypoint <https://github.com/Concordium/concordium-rust-smart-contracts/blob/main/examples/proxy/src/lib.rs#L48>`_).